### ISSUE TITLE
I receive The variable X_position is not defined

### ISSUE THREAD

**Issue by darbid** [section83/MacYTDL#29]
I just downloaded the app and copied it to the applications folder.  I receive the following message.

>The variable X_position is not defined.
>The variable X_position is not defined. (-2753)

MacOS Big Sur Version 11.3

**Comment by section83** [comment]
Ouch.  My apologies. I'm looking into it.

I will need some more detail to be sure I understand the problem. The X and Y position variables open MacYTDL at that position on the desktop. They are only used once and thereafter MacYTDL saves its window position. Can you help with a few questions:

* Is this the first time you have used MacYTDL ?
* Did you see anything else before getting those errors ?
* Do you have an unusual setup or just a standard Mac ? e.g. TeamViewer, VNC, HackInTosh
* Can you find this folder:   Users/[Your Home Folder/Library/Preferences/MacYTDL ?
– If so, please delete the file called "MacYTDL.plist". Then try to open MacYTDL again.

Deleting the plist file should help. When it starts, MacYTDL will create a new plist file with all the default settings (including position).

Please let me know if there was no MacYTDL.plist file or, if deleting it didn't solve the problem.

Many thanks.

Garry

**Comment by darbid** [comment]
Hi Garry;


>Is this the first time you have used MacYTDL ?

Yes. The very first time after putting it in the applications folder. Also from then on.

>Did you see anything else before getting those errors ?

Sorry I do not remember. I don't think so.

>Do you have an unusual setup or just a standard Mac ? e.g. TeamViewer, VNC, HackInTosh

I would guess a pretty standard MacBook Pro and I am not using it in a strange way like with VNC etc. It does have parallels installed.

>Can you find this folder: Users/[Your Home Folder/Library/Preferences/MacYTDL ?

Found and deleted.  This fixed the problem. Now it works.

Below is the text from the deleted file incase this helps.



<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
	<key>Add_Metadata</key>
	<false/>
	<key>Audio_Only</key>
	<false/>
	<key>Auto_Check_YTDL_Update</key>
	<false/>
	<key>Description</key>
	<false/>
	<key>DownloadFolder</key>
	<string>/Users/david/Desktop/</string>
	<key>FileFormat</key>
	<string>Default</string>
	<key>Keep_Remux_Original</key>
	<false/>
	<key>Over-writes allowed</key>
	<false/>
	<key>Remux_Format</key>
	<string>No remux</string>
	<key>Show_Settings_before_Download</key>
	<true/>
	<key>SubTitles</key>
	<false/>
	<key>SubTitles_Embedded</key>
	<false/>
	<key>Subtitles_Format</key>
	<string>best</string>
	<key>Thumbnail_Embed</key>
	<false/>
	<key>Thumbnail_Write</key>
	<false/>
	<key>Verbose</key>
	<false/>
</dict>
</plist>

**Comment by section83** [comment]
It's very odd.  I have found a bug with the X_position variable which would only be noticeable if upgrading from version 1.10 to v1.11.  But it should not affect a first use.

Thank you for the read out of the deleted file.  That will help in the diagnosis.  It should have an entry for "final_position". It's missing some others (e.g. limit_rate and use_proxy). So, there is probably something fundamental that I've got to fix.

Anyway, it should be fine from now on.

Cheers.
